packages/alfa-rules/test/sia-r3/rule.spec.tsx

```diff
@@ -0,0 +1,85 @@
+import { test } from "@siteimprove/alfa-test";
+
+import { Document } from "@siteimprove/alfa-dom";
+
+import R3, { Outcomes } from "../../src/sia-r3/rule";
+
+import { evaluate } from "../common/evaluate";
+import { passed, failed, inapplicable } from "../common/outcome";
+
+test("evaluate() passes one id attribute in the document", async (t) => {
+    const target = <div id="my-div">This is my first element</div>;
+
+    const document = Document.of([target]);
+  
+    t.deepEqual(await evaluate(R3, { document }), [
+      passed(R3, target, {
+        1: Outcomes.HasUniqueId,
+      }),
+    ]);
+  });
+
+test("evaluate() passes unique id attributes in the document context.", async (t) => {
+    const target1 = <div id="my-div1">This is my first element</div>;                        
+    const target2 = <div id="my-div2">This is my second element</div>;
+    const target3 = <svg id="my-div3">This is my third element</svg>;
+
+    const document = Document.of([target1,target2,target3]);
+  
+    t.deepEqual(await evaluate(R3, { document }), [
+      passed(R3, target1, {
+        1: Outcomes.HasUniqueId,        
+      }),
+      passed(R3, target2, {
+        1: Outcomes.HasUniqueId,        
+      }),
+      passed(R3, target3, {
+        1: Outcomes.HasUniqueId,        
+      }),
+    ]);
+  });
+
+test("evaluate() fails unique id attributes in the document context.", async (t) => {
+    const target1 = <div id="label">Name</div>;                        
+    const target2 = <div id="label">City</div>;
+
+    const document = Document.of([target1,target2]);
+  
+    t.deepEqual(await evaluate(R3, { document }), [
+        failed(R3, target1, {
+        1: Outcomes.HasNonUniqueId,        
+      }),
+        failed(R3, target2, {
+        1: Outcomes.HasNonUniqueId,        
+      })      
+    ]);
+  });
+
+  test("evaluate() fails unique id attributes in the document context.", async (t) => {
```

Comment:
```suggestion
  test("evaluate() fails duplicated id attributes on SVG element.", async (t) => {
```
The difference between this test and the previous is that this one is checking an SVG element, not only HTML elements. And so the reason this test is here is to make sure that the rule doesn't ignore SVG elements. And so this should be what the description says 😀